addons/mail/static/src/models/thread_view/thread_view.js

```diff
@@ -113,6 +113,16 @@ function factory(dependencies) {
             }
         }
 
+        /**
+         * Last visible message is scoped to the thread cache.
+         * Needed to be reset when threadCache changes.
+         *
+         * @private
+         */
+        _computeLastVisibleMessage() {
+            return [['unlink']];
+        }
```

Comment:
I really don't like this... This is just a hidden `updateAfter()`. Also problematic when removing dependencies in definition of computed fields.

Reply:
I wouldn't worry about it in stable.
We can/will be able to fix this only if/when we make thread_view_cache model as you suggested in another message.
This is very similar to when we had activities in chatter, the only way to get rid of this crap was to move them to another proper model.

Reply:
Note that removing dependencies is a general problem you will have with all "on change" computes, not just this one.

Reply:
> Note that removing dependencies is a general problem you will have with all "on change" computes, not just this one.

Good point. Reason to make dummy field `onX` to explicitly reset `lastVisibleMessage`.
"on change" computes are in a category of their own, I prefer dedicate struggling with them without other conventional computes.

Reply:
So is that a request for @xdu-odoo to move this compute to a separate `on` field?

Reply:
Yes, sorry if it wasn't clear enough.

@xdu-odoo Don't make `lastVisibleMessage` field computed: instead, make a dummy computed field `onChangeThreadCache` and explicitly unlink `lastVisibleMessage` in compute method.